src/build/help/help.xml

```diff
@@ -542,6 +542,26 @@
                         <example>16KiB=8KiB</example>
                     </config-key>
 
+                    <config-key id="repo-block-size-super" name="Block Incremental Super Block Size">
+                        <summary>Block incremental super block size.</summary>
+
```

Comment:
Another possible approach is to explicitly manage lz4 dictionaries.  In that case, a "superblock" would be defined as a dictionary plus a list of blocks, and each block could be decompressed individually rather than decompressing the entire superblock.  

Did you consider this approach as well? I can see some advantages and some disadvantages, but smaller blocks are more efficient for capturing small changes to a table and explicit dictionaries mean the compression ratios can be the same as for large blocks.

Reply:
I did consider this approach (zstd also supports this), but I was concerned about the complexity of a per compression-method block format. bzip/gzip don't support this feature so we need a generic super block format anyway, which seemed like a good place to start.

I found that 1mb super blocks pretty much negate compression efficiency loss and since they are usually just a few hundred kb compressed fetching them is not too expensive. For object stores (S3, etc) the per request cost is high enough that fetching smaller blocks is not always the advantage that it might seem.

Dictionaries could certainly be implemented in a future release and enabled by something like `repo-block-strategy=dictionary` for the compression methods that support it. lz4 and zstd are not yet available for all pgbackrest builds but that should improved in the future.

Reply:
The header/frame overhead for lz4/zstd is large enough that I think we'd still still want superblocks for very small block sizes (e.g. 8k) even if we used dictionaries. That's just a feeling, though, I have not tested it.

Reply:
Reviewing lz4 at least... looks like the overhead is only along the lines of tens-of-bytes and the same seems to be true for zstd.  Given an 8k block, that seems unlikely to be that much of an issue.  Being able to pull down a much smaller amount of a superblock than the whole thing strikes me as potentially helpful for keeping restores as performant as possible.

On the other hand, I agree that we could implement this later using a configuration option, so at this point, I'd rather have block level incremental in this release than wait for a new approach to be implemented, especially one that can be implemented in the future.

Reply:
And we would likely want to train the dictionary on the super block and then use that dictionary just for the blocks in the super block. I'm worried that training the dictionary on the entire file could lead to poor results. It would certainly require two passes over each file, which I would prefer to avoid.

Reply:
I agree that we would have a separate dictionary per super block.  Note you don't have to train on the entire superblock - it might be reasonable to train on a reasonably sized sample of the superblock.